Kernel/Thread.h

```diff
@@ -471,25 +471,26 @@ class Thread
             VERIFY(move_count > 0);
 
             Vector<BlockerInfo, 4> taken_blockers;
-            taken_blockers.ensure_capacity(move_count);
+            TRY(taken_blockers.try_ensure_capacity(move_count));
             for (size_t i = 0; i < move_count; i++)
-                taken_blockers.append(m_blockers.take(i));
+                TRY(taken_blockers.try_append(m_blockers.take(i)));
             m_blockers.remove(0, move_count);
             return taken_blockers;
         }
 
-        void do_append_blockers(Vector<BlockerInfo, 4>&& blockers_to_append)
+        ErrorOr<void> do_append_blockers(Vector<BlockerInfo, 4>&& blockers_to_append)
```

Comment:
The fix here would be to use intrusive lists for all blockers so we don't have to use Vectors. As you can see we try to use Vectors with reserved capacity so you don't actually dynamically allocate, but it's finicky.